---
uint32 cor_esq
uint32 cor_dir